pages/serverless-functions/troubleshooting/go-function-version-error.mdx

```diff
@@ -0,0 +1,42 @@
+---
+meta:
+  title: Go function build fails due to Go version
+  description: Troubleshoot issues with Scaleway Serverless Functions Go build not working due to version error.
+content:
+  h1: Go function build fails due to Go version
+  paragraph: Troubleshoot issues with Scaleway Serverless Functions Go build not working due to version error.
+tags: serverless functions troubleshooting issue error go version build
+dates:
+  validation: 2025-04-30
+  posted: 2025-04-30
+categories:
+  - serverless
+---
+
+## Problem
+
+My Go Serverless Function build failed with an error message about the wrong version.
+
+## Cause
+
+When deploying a Go function, build step can fail with an error about the version of code versus version of the runtime even if the version seems correct.
+
+This is due to the `go.mod` directive that specify a `X.Y.Z` version where the `Z` version does not match.
+
+This happens by default because in go toolchain, the `go mod init` command now adds the version.
```

Comment:
```suggestion
Go function deployment may fail during the build step with a version mismatch error, even if the runtime version appears to be correct.

This is due to the `go.mod` directive that specifies an `X.Y.Z` version where the `Z` patch version does not match the version of the selected runtime.

This happens by default because the `go mod init` command includes the full Go version (including the patch version), which can lead to this mismatch if your runtime only supports a different patch version.
```